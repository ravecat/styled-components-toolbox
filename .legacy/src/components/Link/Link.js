import React from "react";
import styled from "styled-components";

import { theme } from "common/theme";
import BaseComponent from "components/BaseComponent";

const Link = ({ children, ...rest }) => (
  <Wrapper as="a" {...rest}>
    <span>{children}</span>
  </Wrapper>
);

const Wrapper = styled(BaseComponent).attrs(props => ({
  href: props.href || "#",
  target: props.anchor ? "_self" : "_blank"
}))`
  color: ${theme.darkPrimaryColor};
  text-decoration: none;
`;

export default Link;
