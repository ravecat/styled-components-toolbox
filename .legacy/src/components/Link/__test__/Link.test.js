import React from "react";
import renderer from "react-test-renderer";

import Link from "../index";

it("<Link> renders correctly", () => {
  const tree = renderer.create(<Link>Link</Link>).toJSON();
  expect(tree).toMatchSnapshot();
});
