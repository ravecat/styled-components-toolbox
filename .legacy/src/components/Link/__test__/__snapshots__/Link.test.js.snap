// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> renders correctly 1`] = `
<a
  className="sc-bdVaJa sc-bwzfXH dTSEAF"
  href="#"
  target="_blank"
>
  <span>
    Link
  </span>
</a>
`;
